cli/src/action/registry/api.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright 2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use reqwest::blocking::Client;
+use serde::Deserialize;
+use splinter::registry::Node;
+use std::collections::HashMap;
+
+use crate::action::api::{ServerError, SplinterRestClient};
+use crate::error::CliError;
+
+impl SplinterRestClient {
+    /// Adds a new node to the registry.
+    pub fn add_node(&self, node: &Node) -> Result<(), CliError> {
```

Comment:
We should avoid using Splinter stores' internal structs, which may differ in format from what the CLI sends or the REST API accepts. In fact, this shouldn't work because the `Node` struct shouldn't have the `Serialize` derivation that it does (that's just a result of it being implemented before we started enforcing these guidelines). We should define a struct for this client to represent a node for submission; could probably just re-purpose `NodeResponse`.

Reply:
Changed to use the `RegistryNode` struct

Reply:
`RegistryNode` is good, but it should be defined in the API module. As it is currently, there's a circular dependency: the API module depends on the action module for `RegistryNode`, and the action module depends on the API. Conceptually, the `RegistryNode` struct is a part of the API (it's accepted and returned by its methods).

Reply:
Moved to the API module